app/models/concerns/account_interactions.rb

```diff
@@ -243,6 +243,15 @@ def lists_for_local_distribution
          .where('users.current_sign_in_at > ?', User::ACTIVE_DURATION.ago)
   end
 
+  def followers_hash(url_prefix)
+    key = url_prefix.nil? ? 'local' : url_prefix
+
+    Rails.cache.fetch("followers_hash:#{id}:#{key}") do
+      scope = [nil, 'local'].include?(url_prefix) ? followers.where(domain: nil) : followers.where(Account.arel_table[:uri].matches(url_prefix + '%', false, true))
```

Comment:
I don't super-like that you could have `nil` or `"local"` here as the argument and they mean the same thing. Ideally it should just be one special value, and it could be `nil` like it is for `domain` in other places.

There are potentially thousands of rows being returned here. Even though it's cached, this will mess with memory usage. Ideally, `find_each` should be used. Ruby's `Digest` module supports generating digests from chunks, i.e.:

```ruby
digest = Digest::SHA256.new

scope.find_each { |record| digest.update(record.foo) }

digest.hexdigest
```

The problem is of course that you want the digest to be generated from a canonical, alphabetically sorted order, and our standard pagination order is not that.

Reply:
Regarding the memory consumption... yeah, that could be an issue. I'm not too sure what we can do here since the batch order and the alphabetical order aren't going to be the same... we could at least specialize the query for local VS remote followers in order to select just the required fields.

Reply:
`ORDER BY accounts.username ASC` should cover both local and remote accounts, right? You only ever select either local domain or a specific domain so you don't have to worry about domain sorting. It's not super efficient because it's not an indexed sorting order on that table but it's still more efficient than loading rows into Ruby and sorting afterwards.

Reply:
That being said it wouldn't solve pagination on the rendering side because when loading next page you want to guarantee that you're in the exact place you left off even as the underlying data may shift, and alphabetic sorting does not give that.

Reply:
afaik, `find_each` is based on batch queries, which I think are ordered by `id` and discard any sorting you may have specified in your scope

Reply:
Although I guess there is no specific reason why the rendered collection should be in the same order as the digest? The main thing is that another server could paginate over all records, at that point the digestable order doesn't matter.

Reply:
Oh right. Can we... can we compute the digest in postgres..? Pluck will be a good improvement though.

Reply:
Doing it efficiently in postgres doesn't seem easy either. Ended up doing it another way: XORing individual hashes. Since we are guaranteed to not have duplicates, this should be fine. Furthermore, this avoids issues were sorting would be done with different uses on different implementations/systems.

Reply:
I think using XORed individual hashes solves that problem entirely (as well as the problem of defining an ordering between textual values).